website/static/js/institutionProjectSettings.js

```diff
@@ -5,102 +5,174 @@ var Raven = require('raven-js');
-            message: 'You are about to remove affiliation with ' + item.name + ' from this project. ' + item.name + ' branding will not longer appear on this project, and the project will not be discoverable on the ' + item.name + ' landing page.',
-            callback: function (confirmed) {
-                if (confirmed) {
-                    self._clearInst(item);
-                }
-            },
-            buttons:{
-                confirm:{
-                    label:'Remove affiliation'
-                }
-            }
-        });
+        self.needsWarning((self.userInstitutionsIds.indexOf(item.id) === -1));
+        self.isAddInstitution(false);
+        if (self.hasChildren()) {
+            self.modifyChildrenDialog(item);
+        }
+        else {
+            return self._modifyInst(item);
+        }
     };
-    self._clearInst = function(item) {
-        var url = data.apiV2Prefix + 'nodes/' + data.node.id + '/relationships/institutions/';
+
+    self._modifyInst = function(item) {
+        var index;
+        var url = data.apiV2Prefix + 'institutions/' + item.id + '/relationships/nodes/';
+        var ajaxJSONType = self.isAddInstitution() ? 'POST': 'DELETE';
+        var nodesToModify = [];
+        self.loading(true);
+        if (self.modifyChildren()) {
+            for (var node in self.nodesOriginal()) {
+                nodesToModify.push({'type': 'nodes', 'id': self.nodesOriginal()[node].id});
+            }
+        }
+        else {
+            nodesToModify.push({'type': 'nodes', 'id': self.nodeParent});
+        }
         return $osf.ajaxJSON(
-            'DELETE',
+            ajaxJSONType,
             url,
             {
                 isCors: true,
                 data: {
-                     'data': [{'type': 'institutions', 'id': item.id}]
+                     'data': nodesToModify
                 },
                 fields: {xhrFields: {withCredentials: true}}
             }
-        ).done(function (response) {
-            var indexes = self.affiliatedInstitutions().map(function(each){return each.id;});
-            var removed = self.affiliatedInstitutions().splice(indexes.indexOf(item.id), 1)[0];
-            if ($.inArray(removed.id, self.userInstitutionsIds) >= 0){
+        ).done(function () {
+            if (self.isAddInstitution()) {
+                index = self.availableInstitutionsIds().indexOf(item.id);
+                var added = self.availableInstitutions().splice(index, 1)[0];
+                self.affiliatedInstitutions().push(added);
+            }
+            else {
+                index = self.affiliatedInstitutionsIds().indexOf(item.id);
+                var removed = self.affiliatedInstitutions().splice(index, 1)[0];
                 self.availableInstitutions().push(removed);
-                self.availableInstitutions(self.availableInstitutions());
             }
+            self.availableInstitutions(self.availableInstitutions());
```

Comment:
Discussed on Flowdock: to correctly modify observableArrays, you need to make sure to not call them

```js
self.affiliInsti.push(...)   // not self.affiliInsti().push(...)
self.affiliInsti.splice(...) // not self.affiliInst().splice(...)
```